File: image/net9/Controllers/ImageController.cs
using Microsoft.AspNetCore.Mvc;
using net9.Models.Requests;
using net9.Services;

namespace net9.Controllers;

[ApiController]
// [Route("api/v1/image")]
[Route("image")]
public class ImageController: ControllerBase
{
    private readonly IImageService _imageservice;
    public ImageController(IImageService imageService)
    {
        _imageservice = imageService;
    }
    
    [HttpPost]
    public IActionResult CheckImage([FromBody] ImageRequest imageRequest)
    {
        string response = _imageservice.CheckImage(imageRequest);
        return StatusCode(200, response);
    }
}

File: image/rust/src/controllers/image_controller.rs
use std::sync::Arc;

use axum::{extract::State, http::StatusCode, response::IntoResponse, Json};

use crate::{models::requests::image_request::ImageRequest, services::image_service::ImageService, states::app_state::AppState};

pub async fn check_image(
    State(state): State<Arc<AppState>>,
    Json(image_request): Json<ImageRequest>
) -> impl IntoResponse {
    let response = state.image_service.check_image(image_request).await;
    (StatusCode::OK, response)
}